### DISCUSSION TITLE
Is it possible to assign different speed of vehicles for different time windows of a day ?

### DISCUSSION THREAD

**Discussion by AKA2501** [Project-OSRM/osrm-backend#6808] [Q&A]
I am using a routing optimizing machine called VROOM

My architecture is like the following 

Where i have osrm installed using docker and Vroom runs over it helping in processing and sending a request then to OSRM for its routing matrix and profile.
Currently we get only a single speed value for the entire process of routing.
The current question i have is that 
Is there a possibility of using different speed values or ranges during different times of the day and if so yes what changes or necessary modifications i must make for the same

**Comment by nilsnolde** [comment]
OSRM can't track time, so it can't deal with different time in a week. The only thing you could do, and I know people doing that too, is to run multiple instances of OSRM, one for average congestion, one for medium congestion and one free-flow for example. Then you can mix & match. Do note though, that VROOM can't deal with time dependence either, so on the optimization side you're definitely stuck with one of those instances.

Valhalla can deal with time dependence, it can track time & corresponding speeds in 5 mins intervals. For that you obviously need the appropriate data from TomTom or HERE and you'd need to convert that data to Valhlalla's proprietary traffic format.

**Comment by AKA2501** [comment]
Can you please elaborate on the process of running multiple instances of OSRM depending on the Congestion and also how can define the congestion property while setting up a different instance.
